gitlab/client.py

```diff
@@ -818,7 +825,28 @@ def http_list(
 
         if page or as_list is True:
             # pagination requested, we return a list
-            return list(GitlabList(self, url, query_data, get_next=False, **kwargs))
+            gl_list = GitlabList(self, url, query_data, get_next=False, **kwargs)
+            items = list(gl_list)
+            if page is None and get_all is None and gl_list.per_page is not None:
```

Comment:
We're first checking for `page` and then checking if it is None - maybe an early return for at least that branch (or more scenarios) would make the if expressions a bit simpler here?

Better yet, invert the logic to flatten the entire thing:
1. if as_list is false, return generator
2. if page, return list
3. ...maybe more
3. otherwise do the complex logic below

Reply:
Thanks. Good idea. I split it up.